NLog.Web.AspNetCore/LayoutRenderers/AspNetRequestIpLayoutRenderer.cs

```diff
@@ -16,18 +19,52 @@ namespace NLog.Web.LayoutRenderers
     [LayoutRenderer("aspnet-request-ip")]
     public class AspNetRequestIpLayoutRenderer : AspNetLayoutRendererBase
     {
+        private const string ForwardedForHeader = "X-Forwarded-For";
+
+        /// <summary>
+        /// Gets or sets whether the renderer should check value of X-Forwarded-For header
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+        public bool CheckForwardedForHeader { get; set; }
+
         /// <summary>
         /// Render IP
         /// </summary>
         protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
         {
+            var forwardedIp = string.Empty;
             var httpContext = HttpContextAccessor.HttpContext;
+
 #if !ASP_NET_CORE
+            if (CheckForwardedForHeader)
+            {
+                var forwardedHeader = httpContext.TryGetRequest().Headers[ForwardedForHeader];
```

Comment:
Instead of calling httpContext.TryGetRequest() twice, then do it once reuse it for `Headers[ForwardedForHeader]` and `ServerVariables["REMOTE_ADDR"]` (Maybe also handle null value from TryGetRequest)

Reply:
Done